aspnetcore/signalr/configuration.md

```diff
@@ -0,0 +1,126 @@
+---
+title: ASP.NET Core SignalR configuration
+author: rachelappel
+description: Configure ASP.NET Core SignalR Apps
+manager: wpickett
+monikerRange: '>= aspnetcore-2.1'
+ms.author: rachelap
+ms.date: 06/30/2018
+ms.prod: asp.net-core
+ms.technology: aspnet
+ms.topic: article
+uid: signalr/configuration
+---
+
+# ASP.NET Core SignalR configuration
+
+## JSON/MessagePack serialization options
+
+ASP.NET Core SignalR supports two protocols for encoding messages, [JSON](https://www.json.org/) and [MessagePack](https://msgpack.org/index.html). Each protocol has serialization configuration options.
+
+JSON serialization can be configured on the server using the `AddJsonProtocol` extension method. The `AddJsonProtocol` method takes a delegate that receives an `options` object. The `PayloadSerializerSettings` property on that object is a JSON.NET `JsonSerializerSettings` object that can be used to configure serialization of arguments and return values. See the [JSON.NET Documentation](https://www.newtonsoft.com/json/help/html/Introduction.htm) for more details.
+
+As an example, you can configure the serializer to use PascalCase names instead of the default camelCase names using the following code:
+
+[!code-csharp[Startup config](configuration/sample/config-startup.cs?range=1-5)]
+
+In the .NET client, the same `AddJsonHubProtocol` extension method exists on [HubConnectionBuilder](/dotnet/api/microsoft.aspnetcore.signalr.client.hubconnectionbuilder). The `Microsoft.Extensions.DependencyInjection` namespace must be imported to resolve the extension method:
+
+[!code-csharp[HubConnectionBuilder](configuration/sample/addhubjsonprotocol.cs)]
+
+### MessagePack Serialization Options
+
+MessagePack serialization can be configured by providing a delegate to the [AddMessagePackProtocol](/dotnet/api/microsoft.extensions.dependencyinjection.msgpackprotocoldependencyinjectionextensions.addmessagepackprotocol) call. See [MessagePack in SignalR](xref:signalr/messagepack) for more details.
+
+> [!NOTE]
+> It's not possible to configure JSON or MessagePack serialization in the JavaScript client at this time.
+
+### Configure [HubOptions](/dotnet/api/microsoft.aspnetcore.signalr.huboptions) for SignalR
```

Comment:
We should make all these headers a little more consistent. How about we remove the type names from the headers, and leave those for the prose below.

Also, I think we can just merge this with the next section (`HttpConnectionDispatcherOptions`) as a "Configuring Server Options" section. The section below on the Client has multiple types and I think it works fine there.